digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/integration/Viite_13_218_spec.scala

```diff
@@ -1174,7 +1174,8 @@ class Viite_13_218_spec extends FunSuite with Matchers with BeforeAndAfter {
         check_two_track_continuous(two_track_nonterminated_sources)
         check_two_track_continuous(two_track_nonterminated_targets)
 
-        projectService_db.updateRoadwaysAndLinearLocationsWithProjectLinks(ProjectState.Saved2TR, projectSaved.id)
+        projectDAO.updateProjectStatus(projectSaved.id, ProjectState.UpdatingToRoadNetwork)
+        projectService_db.updateRoadwaysAndLinearLocationsWithProjectLinks(projectSaved.id)
```

Comment:
In the same transaction?

Reply:
Elaborate to a whole-sentenced question, please? :D

Trying to answer without understanding the whole question:

1. runWithRollback at line ~280 -> are within the same transaction?

2. ProjectStatus must be updated before modified updateRoadwaysAndLinearLocationsWithProjectLinks is called.
(Refactored the ProjectStatus away from the args, as it made it possible for the caller to use a random ProjectState to "check, that project state is correct", which kind of digs the groud off below the whole check...)